daemon/logger/loggerutils/logfile.go

```diff
@@ -2,32 +2,47 @@ package loggerutils
 
 import (
 	"bytes"
+	"compress/gzip"
 	"context"
+	"encoding/json"
 	"fmt"
 	"io"
 	"os"
 	"strconv"
+	"strings"
 	"sync"
 	"time"
 
 	"github.com/docker/docker/daemon/logger"
 	"github.com/docker/docker/daemon/logger/loggerutils/multireader"
 	"github.com/docker/docker/pkg/filenotify"
+	"github.com/docker/docker/pkg/jsonmessage"
+	"github.com/docker/docker/pkg/pools"
 	"github.com/docker/docker/pkg/pubsub"
 	"github.com/docker/docker/pkg/tailfile"
 	"github.com/fsnotify/fsnotify"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
+const tmpLogfileSuffix = ".tmp"
+
+// RotateFileMetadata is a metadata of the gzip header of the compressed log file
+type RotateFileMetadata struct {
+	LastTime string `json:"lastTime,omitempty"`
```

Comment:
Can we use `time.Time` here?

Reply:
This could also likely do well to have a first and last timestamp.

Reply:
>Can we use time.Time here?

Will do.
> This could also likely do well to have a first and last timestamp.
In general, the last timestamp of previous log file is the first timestramp the current log file, so I think last timestamp is enough.